pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -113,12 +113,12 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerConfig(container *v1.C
 		// for container level cgroup.
 		memoryHigh := int64(0)
 		if memoryLimit != 0 {
```

Comment:
We need to add a check before this if-else for guaranteed pods.
pseudo code:
if !guaranteed pods {
  set memory.high
}